core/storage/trie/polkadot_trie/polkadot_trie_impl.cpp

```diff
@@ -221,6 +220,44 @@ namespace kagome::storage::trie {
     return nullptr;
   }
 
+  outcome::result<std::list<std::pair<PolkadotTrieImpl::BranchPtr, uint8_t>>>
+  PolkadotTrieImpl::getPath(NodePtr parent,
+                            const common::Buffer &key_nibbles) const {
+    using Path = std::list<std::pair<PolkadotTrieImpl::BranchPtr, uint8_t>>;
+    using T = PolkadotNode::Type;
+    if (parent == nullptr) {
+      return Path{};
+    }
+    switch (parent->getTrieType()) {
+      case T::BranchEmptyValue:
+      case T::BranchWithValue: {
+        auto length = getCommonPrefixLength(parent->key_nibbles, key_nibbles);
+        if (parent->key_nibbles == key_nibbles || key_nibbles.empty()) {
+          auto found_leaf =
+              std::make_shared<LeafNode>(parent->key_nibbles, parent->value);
+          return Path{};
+        }
+        if ((parent->key_nibbles.subbuffer(0, length) == key_nibbles)
+            && key_nibbles.size() < parent->key_nibbles.size()) {
+          return Path{};
+        }
+        auto parent_as_branch = std::dynamic_pointer_cast<BranchNode>(parent);
+        OUTCOME_TRY(n, retrieveChild(parent_as_branch, key_nibbles[length]));
+        OUTCOME_TRY(path, getPath(n, key_nibbles.subbuffer(length + 1)));
+        path.push_front({parent_as_branch, key_nibbles[length]});
+        return std::move(path);
```

Comment:
i'm not sure, but it's generally not required ot move value beign returned, compiler does it by default. probably clang-tidy will write a warning here.

Reply:
however the real return value is outcome::result, so it can be different in this case.

Reply:
No, we don't need it. NRVO does not work here, because imlicit return exists it other OUTCOME_TRY

Reply:
So, as NRVO doesn't work I take it we do need an explicit move here?

Reply:
Absolutly! =)